source/qml/Library/RendererQml/TextInputRender.cpp

```diff
@@ -381,82 +118,72 @@ void TextInputElement::addInlineActionMode()
-const std::string TextInputElement::getColorFunction()
-{
-    const auto textConfig = mContext->GetRenderConfig()->getInputTextConfig();
-    std::string colorFunction = RendererQml::Formatter() << "function colorChange(colorItem,focusItem,isPressed) {\n"
-        "if (isPressed && !focusItem.showErrorMessage) {\n"
-        "colorItem.color = " << mContext->GetHexColor(textConfig.backgroundColorOnPressed) << "\n"
-        "}\n"
-        "else {\n"
-        "colorItem.color = focusItem.showErrorMessage ? " << mContext->GetHexColor(textConfig.backgroundColorOnError) << " : focusItem.activeFocus ? " << mContext->GetHexColor(textConfig.backgroundColorOnPressed) << " : focusItem.hovered ? " << mContext->GetHexColor(textConfig.backgroundColorOnHovered) << " : " << mContext->GetHexColor(textConfig.backgroundColorNormal) << "\n"
-        "}\n"
-        "}\n";
-    return colorFunction;
-}
+            mTextinputColElement->Property("isIconLeftOfTitle", context->GetConfig()->GetActions().iconPlacement == AdaptiveCards::IconPlacement::LeftOfTitle ? "true" : "false");
+            mTextinputColElement->Property("escapedTitle", RendererQml::Formatter() << "String.raw`" << RendererQml::Utils::getBackQuoteEscapedString(action->GetTitle()) << "`");
+            // always sending as false not supported 
+            mTextinputColElement->Property("isShowCardButton", "false");
+            mTextinputColElement->Property("isActionSubmit", action->GetElementTypeString() == "Action.Submit" ? "true" : "false");
+            mTextinputColElement->Property("isActionOpenUrl", action->GetElementTypeString() == "Action.OpenUrl" ? "true" : "false");
+            mTextinputColElement->Property("isActionToggleVisibility", action->GetElementTypeString() == "Action.ToggleVisibility" ? "true" : "false");
 
-std::string TextInputElement::getAccessibleName(std::shared_ptr<RendererQml::QmlTag> uiTextInput)
-{
-    std::ostringstream accessibleName;
-    std::ostringstream labelString;
-    std::ostringstream errorString;
-    std::ostringstream placeHolderString;
+            if (!action->GetIconUrl().empty())
+            {
+                mTextinputColElement->Property("hasIconUrl", "true");
+                mTextinputColElement->Property("imgSource", RendererQml::AdaptiveCardQmlRenderer::GetImagePath(context, action->GetIconUrl()), true);
+            }
+            std::string selectActionId = "";
+            if (action->GetElementTypeString() == "Action.OpenUrl")
+            {
+                auto openUrlAction = std::dynamic_pointer_cast<AdaptiveCards::OpenUrlAction>(action);
+                selectActionId = openUrlAction->GetUrl();
 
-    if(mContext->GetRenderConfig()->isAdaptiveCards1_3SchemaEnabled())
-    {
-        if (!mTextinput->GetLabel().empty())
-        {
-            labelString << "accessibleName += String.raw`" << mEscapedLabelString << ". `;";
-        }
+            }
+            else if (action->GetElementTypeString() == "Action.ToggleVisibility")
```

Comment:
Can we refactor these into separate functions in AdaptiveCardQmlRenderer.cpp so that we can use the same for all elements

Reply:
Done